### ISSUE TITLE
POST /keys/upload schema and example don't match.

### ISSUE THREAD

**Issue by Dominaezzz** [matrix-org/matrix-doc#2256]
So for this property, https://github.com/matrix-org/matrix-doc/blob/03641d9e48c3e3e73ff8b3c851e2338fb8ceea87/api/client-server/keys.yaml#L75-L80

One of the examples are,
https://github.com/matrix-org/matrix-doc/blob/03641d9e48c3e3e73ff8b3c851e2338fb8ceea87/api/client-server/keys.yaml#L86-L90

The schema and example doesn't match so one of them is wrong. (The doc seems to insist the example is wrong).

TL;DR
The schema says, `Map<String, String>` but the example is `Map<String, Map<String, String>>`.

**Comment by turt2live** [comment]
The schema does say it is a `Map<String, Map<String, String>>` due to the use of `additionalProperties`, it's just poorly represented.

Line 69 identifies the whole thing as an object (first `Map`), then line 79 says it consists of a `Map<String, String>`, so when put together it's `Map<String, Map<String, String>>`.

**Comment by Dominaezzz** [comment]
> then line 79 says it consists of a Map<String, String>

Yes, line 79 consists of `Map<String, String>` but the example of it is `Map<String, Map<String, String>>.`

**Comment by turt2live** [comment]
Right, the signatures property itself is a `Map<String, *>` because of line 69. Line 79 makes the `*` a `Map<String, String>`.

**Comment by Dominaezzz** [comment]
:confused: Don't you mean because of line 76?

Also, if `signatures` itself is a `Map<String, String>`, why does the doc say "Mapped from user ID to signature object." (as supposed to "signature string")?

**Comment by Dominaezzz** [comment]
If `signatures` is a map of user ID to signature object. (Assuming the doc is correct).

Taking one of the examples provided.
```
"signatures": { 
  "@alice:example.com": { 
    "ed25519:JLAFKJWSCS": "FLWxXqGbwrb8SM3Y795eB6OA8bwBcoMZFXBqnTn58AYWZSqiD45tlBVcDa2L7RwdKXebW/VzDlnfVJ+9jok1Bw" 
  } 
} 
```

Am I correct in say that `@alice:example.com` is a user ID and
```
{ 
   "ed25519:JLAFKJWSCS": "FLWxXqGbwrb8SM3Y795eB6OA8bwBcoMZFXBqnTn58AYWZSqiD45tlBVcDa2L7RwdKXebW/VzDlnfVJ+9jok1Bw" 
}
```
is the mapped signature **object**?

**Comment by turt2live** [comment]
yes
